src/droplet.F

```diff
@@ -1791,15 +1846,58 @@ subroutine droplet_diag(rtime,nrec,zh,zf,pdata,pdata_locind)
       if ( myid == 0 ) then
          call mpi_reduce(MPI_IN_PLACE,partnum,nk,MPI_REAL, &
```

Comment:
I would like to encourage the use of MPI_Allreduce, having a large number of duplicate mpi_reduce calls does not help code complexity.

Reply:
Just a comment here: in subsequent stages of this overall pull request, all of the "1D" stats are combined into a single array called `droplet_diag_1`. So in a future sub-PR there will only be a single mpi_reduce call for an nnz-by-20 array, which contains all of them at once. Same idea as the `droplet_diag_0` array. So maybe just hold off on this comment until you see how it looks